### ISSUE TITLE
Pack install 500 server error

### ISSUE THREAD

**Issue by vakker** [SWI-Prolog/swipl-devel#482]
Hi

I tried installing a package with `pack_install`, e.g. `pack_install(hello).` from [the doc](https://www.swi-prolog.org/pack/list), but it fails with:
```
% Contacting server at http://www.swi-prolog.org/pack/query ...
ERROR: url `'https://www.swi-prolog.org/pack/query'' does not exist (status(500,Internal Server Error))
ERROR: In:
ERROR:   [18] throw(error(existence_error(url,'https://www.swi-prolog.org/pack/query'),context(_17632,...)))
ERROR:   [16] http_open:try_http_proxy(direct,[uri('http://www.swi-prolog.org/pack/query'),...|...],_17668,[post(...),...]) at /usr/lib/swi-prolog/library/http/http_open.pl:425
ERROR:   [14] <meta call>
ERROR:   [13] '$sig_atomic'(prolog_pack:http_open('http://www.swi-prolog.org/pack/query',_17762,...)) <foreign>
ERROR:   [12] setup_call_catcher_cleanup(prolog_pack:http_open('http://www.swi-prolog.org/pack/query',_17806,...),prolog_pack:read_reply(_17818,_17820,_17822),_17792,prolog_pack:close(_17832)) at /usr/lib/swi-prolog/boot/init.pl:469
ERROR:   [10] prolog_pack:query_pack_server(locate(hello),_17864,[]) at /usr/lib/swi-prolog/library/prolog_pack.pl:1733
ERROR:    [9] prolog_pack:pack_default_options(hello,hello,[],_17906) at /usr/lib/swi-prolog/library/prolog_pack.pl:516
ERROR:    [8] prolog_pack:pack_install(hello) at /usr/lib/swi-prolog/library/prolog_pack.pl:457
ERROR:    [7] <user>
ERROR: 
ERROR: Note: some frames are missing due to last-call optimization.
ERROR: Re-run your program in debug mode (:- debug.) to get more detail.
```

Checking the page http://www.swi-prolog.org/pack/query manually in a browser gives:
```
Internal server error

url `'http://eu.swi-prolog.org/pack/query'' does not exist (status(500,Internal Server Error))
In:
[20] throw(error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_3928,...)))
[18] http_open:try_http_proxy(direct,[uri('http://eu.swi-prolog.org/pack/query'),...|...],_3964,[method(get),...|...]) at /home/jan/lib/swipl/library/http/http_open.pl:425
[16] plweb_proxy:proxy(get,"http://eu.swi-prolog.org",[protocol(http),...|...],[request_headers(...)]) at /home/jan/src/plweb/proxy.pl:80
[13] pack:pack_query([protocol(http),...|...]) at /home/jan/src/plweb/pack.pl:81
[12] http_dispatch:call_action(pack:pack_query,[protocol(http),...|...]) at /home/jan/lib/swipl/library/http/http_dispatch.pl:1007
[10] time:run_alarm_goal('$alarm'(11580435498180),http_dispatch:call_action(...,...,...)) at /home/jan/lib/swipl/library/time.pl:145
[9] setup_call_catcher_cleanup(time:alarm(300,...,...,...),time:run_alarm_goal(...,...),_842,time:remove_alarm_notrace(...)) at /home/jan/lib/swipl/boot/init.pl:468
[5] httpd_wrapper:call_handler(http_dispatch:time_limit_action(...,...,...),1817604,-) at /home/jan/lib/swipl/library/http/http_wrapper.pl:314
[4] catch(httpd_wrapper:call_handler(...,1817604,-),error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_988,...)),httpd_wrapper:true) at /home/jan/lib/swipl/boot/init.pl:386
[3] httpd_wrapper:handler_with_output_to(http_dispatch:time_limit_action(...,...,...),1817604,-,current_output,error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_1064,...))) at /home/jan/lib/swipl/library/http/http_wrapper.pl:297

Note: some frames are missing due to last-call optimization.
Re-run your program in debug mode (:- debug.) to get more detail.
SWI-Prolog httpd at swi-prolog.osuosl.org 
```

**Comment by erlanger** [comment]
Hi @vakker,

   This has been fixed in latest dev version and in stable 8.0.3.

**Comment by vakker** [comment]
Yes it's fixed now.
For the record, some packages are broken, e.g.:
```
?- pack_install(hello).
% Contacting server at https://www.swi-prolog.org/pack/query ... ok
Install hello@0.0.1 from http://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip Y/n? 
ERROR: url `'https://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip'' does not exist (status(404,Not Found))
ERROR: In:
ERROR:   [19] throw(error(existence_error(url,'https://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip'),context(_28290,...)))
ERROR:   [17] http_open:try_http_proxy(direct,[uri('http://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip'),...|...],_28326,[cert_verify_hook(...)]) at /usr/lib/swi-prolog/library/http/http_open.pl:425
ERROR:   [15] <meta call>
ERROR:   [14] '$sig_atomic'(prolog_pack:http_open('http://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip',_28414,...)) <foreign>
ERROR:   [13] setup_call_catcher_cleanup(prolog_pack:http_open('http://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip',_28458,...),prolog_pack:setup_call_cleanup(...,...,...),_28444,prolog_pack:close(_28484)) at /usr/lib/swi-prolog/boot/init.pl:469
ERROR:   [11] prolog_pack:pack_install_from_url(http,'http://files.dropbox.com/u/1397729/projects/pack/hello-0.0.1.zip','/home/vince/lib/swipl/pack',hello,[inquiry(true),...|...]) at /usr/lib/swi-prolog/library/prolog_pack.pl:921
ERROR:    [7] <user>
ERROR: 
ERROR: Note: some frames are missing due to last-call optimization.
ERROR: Re-run your program in debug mode (:- debug.) to get more detail.
```

while others are fine, e.g.:
```
?- pack_install(ffi).
% Contacting server at https://www.swi-prolog.org/pack/query ...
 ok
Install ffi@0.1 from GIT at http://github.com/JanWielemaker/ffi.git Y/n? 
% Cloning into '/home/user/lib/swipl/pack/ffi'...
Warning: warning: redirecting to https://github.com/JanWielemaker/ffi.git/
% Contacting server at https://www.swi-prolog.org/pack/query ... ok
% "ffi.git" was downloaded 1 times
Package:                ffi
Title:                  Dynamically call C functions
Installed version:      0.1
Author:                 Jan Wielemaker <J.Wielemaker@vu.nl>
Maintainer:             Jan Wielemaker <J.Wielemaker@vu.nl>
Packager:               Jan Wielemaker <J.Wielemaker@vu.nl>
Home page:              https://github.com/JanWielemaker/ffi.git
Download URL:           https://github.com/JanWielemaker/ffi/archive/V*.zip
Run post installation scripts for pack "ffi" Y/n? 
% checking for gcc... 
...
```

**Comment by kaaloo** [comment]
Hello, I'm seeing this behavior now when going to http://www.swi-prolog.org/pack/query

Internal server error
url `'http://eu.swi-prolog.org/pack/query'' does not exist (status(500,Internal Server Error))
In:
[20] throw(error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_4728,...)))
[18] http_open:try_http_proxy(direct,[uri('http://eu.swi-prolog.org/pack/query'),...|...],_4764,[method(get),...|...]) at /home/jan/lib/swipl/library/http/http_open.pl:459
[16] plweb_proxy:proxy(get,"http://eu.swi-prolog.org",[protocol(http),...|...],[request_headers(...)]) at /home/jan/src/plweb/proxy.pl:80
[13] pack:pack_query([protocol(http),...|...]) at /home/jan/src/plweb/pack.pl:81
[12] http_dispatch:call_action(pack:pack_query,[protocol(http),...|...]) at /home/jan/lib/swipl/library/http/http_dispatch.pl:1030
[10] time:run_alarm_goal('$alarm'(94785777317344),http_dispatch:call_action(...,...,...)) at /home/jan/lib/swipl/library/time.pl:147
[9] setup_call_catcher_cleanup(time:alarm(300,...,...,...),time:run_alarm_goal(...,...),_1030,time:remove_alarm_notrace(...)) at /home/jan/lib/swipl/boot/init.pl:619
[5] httpd_wrapper:call_handler(http_dispatch:time_limit_action(...,...,...),3623,-) at /home/jan/lib/swipl/library/http/http_wrapper.pl:323
[4] catch(httpd_wrapper:call_handler(...,3623,-),error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_1176,...)),httpd_wrapper:true) at /home/jan/lib/swipl/boot/init.pl:537
[3] httpd_wrapper:handler_with_output_to(http_dispatch:time_limit_action(...,...,...),3623,-,current_output,error(existence_error(url,'http://eu.swi-prolog.org/pack/query'),context(_1252,...))) at /home/jan/lib/swipl/library/http/http_wrapper.pl:306

Note: some frames are missing due to last-call optimization.
Re-run your program in debug mode (:- debug.) to get more detail.

SWI-Prolog httpd at swi-prolog.osuosl.org

**Comment by JanWielemaker** [comment]
See above in the thread

> This has been fixed in latest dev version and in stable 8.0.3.
> Please close after you verify. Thanks!

**Comment by kaaloo** [comment]
Oh drats, sorry about the noise! Indeed the version on my distribution was outdated.  I'm now installing from the PPA.
